dockstore-webservice/src/main/java/io/dockstore/webservice/resources/ElasticsearchConsistencyHealthCheck.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2024 OICR and UCSC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *           http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.dockstore.webservice.resources;
+
+import com.codahale.metrics.health.HealthCheck;
+import io.dockstore.webservice.helpers.ElasticSearchHelper;
+import io.dockstore.webservice.jdbi.AppToolDAO;
+import io.dockstore.webservice.jdbi.BioWorkflowDAO;
+import io.dockstore.webservice.jdbi.EntryDAO;
+import io.dockstore.webservice.jdbi.NotebookDAO;
+import io.dockstore.webservice.jdbi.ToolDAO;
+import io.dropwizard.hibernate.UnitOfWork;
+import java.io.IOException;
+import org.apache.http.HttpStatus;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.RestHighLevelClient;
+import org.elasticsearch.client.core.CountRequest;
+import org.elasticsearch.client.core.CountResponse;
+
+public class ElasticsearchConsistencyHealthCheck extends HealthCheck  {
+    private ToolDAO toolDAO;
+    private BioWorkflowDAO bioWorkflowDAO;
+    private AppToolDAO appToolDAO;
+    private NotebookDAO notebookDAO;
+
+    public ElasticsearchConsistencyHealthCheck(ToolDAO toolDAO, BioWorkflowDAO bioWorkflowDAO, AppToolDAO appToolDAO, NotebookDAO notebookDAO) {
+        this.toolDAO = toolDAO;
+        this.bioWorkflowDAO = bioWorkflowDAO;
+        this.appToolDAO = appToolDAO;
+        this.notebookDAO = notebookDAO;
+    }
+
+    private long countElasticsearchDocuments(String index) throws IOException {
+        RestHighLevelClient client = ElasticSearchHelper.restHighLevelClient();
+        CountRequest countRequest = new CountRequest(index);
+        CountResponse countResponse = client.count(countRequest, RequestOptions.DEFAULT);
+        if (countResponse.status().getStatus() != HttpStatus.SC_OK) {
+            throw new RuntimeException("Non-OK response to Elasticsearch request");
```

Comment:
Should this be `CustomWebApplicationException` with an appropriate error code?

Reply:
When the health check system we use runs each health check, it catches any exception that the health check throws and maps it to an unhealthy result.  These results are then mapped to a response by our endpoint code.   If it's a `CustomWebApplicationException`, that implies that it's gonna emerge from the webservice with the attached code, which isn't going to happen.  So, I'd probably lean against it.  I'd prefer that we throw a specialization of `RuntimeException` that was more specific to the particular error, but I couldn't find one.

Reply:
Makes sense, maybe https://docs.oracle.com/en/java/javase/17/docs/api/jdk.jdi/com/sun/jdi/InternalException.html

Reply:
`InternalException` is meant to mirror an internal JDK error, see the package documentation: https://docs.oracle.com/en/java/javase/17/docs/api/jdk.jdi/com/sun/jdi/package-summary.html

Reply:
Could also declare your own